tests/e2e/no_hci_mesh_rwx_singlevc_topology_disruptive.go

```diff
@@ -0,0 +1,726 @@
+			}
+		}
+	})
+
+	/*
+	   TESTCASE-1
+	   Bring down full site cluster-2
+	   SC → Immediate Binding Mode, default values
+
+	   Steps:
+	   1. Create SC with Imemdiate Binding mode and set SC with default values
+	   2. Create around 10 PVCs with RWX access mode.
+	   3. Wait for all PVCs to reach Bound state.
+	   4. Create deployment Pods for each PVC with replica count 7 - specify nodeSelectorTerms for any one deployment set.
+	   5. Wait for Pods to reach running state
+	   6. Verify that all the Pods are spread across all the AZ as specified in the node selector terms.
+	   7. Bring Az2 site completely down.
+	   10. While one site is completely down, In between, Scale up first 3 deployment sets to 10 replica count.
+	   11. Wait for some time and make sure all 10 replicas Pods for the deployment sets are
+	   in running state and new PVCs to be in Bound state.
+	   12. Scale down next 3 deployment sets to 4 replicas
+	   13. Wait for some time and verify that scale down operation went successful.
+	   14. Note - verify if we are able to read or write on to the volumes when site is down
+	   15. Create statefulset with replica count 3 with pvc created with rwx access mode
+	   16. Verify sts pods to be in running/ready and pvc to be in Bound state.
+	   17. Create new pvc and attach it to a standalone pod post site recovery.
+	   18. Bring up the site which was down.
+	   19. Wait for testbed to be back to normal state.
+	   20. Verify the k8s nodes status and CSI Pods status.
+	   21. Verify that all the workload Pods are in up and running state after site recovery.
+	   22. Perform scaling operation again on any deployment set.
+	   23. Verify scaling operation went smooth.
+	   24. Make sure K8s cluster  is healthy
+	   25. Perform cleanup by deleting Pods, PVCs and SC
+	*/
+
+	ginkgo.It("When full site Az2 is down", ginkgo.Label(p1, file, vanilla, level5, level2,
+		newTest, disruptive), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var podsNodeSelectorterm *v1.PodList
+		var podList []*v1.Pod
+		var deploymentList []*appsv1.Deployment
+		var service *v1.Service
+
+		// deployment pod and pvc count
+		stsReplicas = 3
+		replica = 7
+		createPvcItr = 10
+		createDepItr = 1
+		noPodsToDeploy = 2
+
+		ginkgo.By(fmt.Sprintf("Creating Storage Class with access mode %q and fstype %q with "+
+			"no allowed topology specified", accessmode, nfs4FSType))
+		storageclass, pvclaims, err := createStorageClassWithMultiplePVCs(client, namespace, labelsMap,
+			scParameters, diskSize, nil, bindingModeImm, false, accessmode, defaultNginxStorageClassName,
+			nil, createPvcItr, false, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		defer func() {
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
```

Comment:
Is this needed here?? no sts has been created above.

Reply:
It is created down. I am doing stateful set cleanup at the last after performing cleanup of all PVCs, deployment Pods and standalone pods so that in case if any stale entry left for PVC or PV that should be handled by this DeleteAllStatefulSets util. Inside this util, we have a logic of performing all the PVCs and PVs cleanup